examples/build.sh

```
@@ -1,55 +1,89 @@
-echo "$BAL_DESTINATION_DIR"
-echo "$BAL_SOURCE_DIR"
+echo "Source Directory: $BAL_SOURCE_DIR"
+echo "Destination Directory: $BAL_DESTINATION_DIR"
 
-# Loop through examples in the examples directory
-find "$BAL_EXAMPLES_DIR" -type f -name "*.bal" | while read -r BAL_EXAMPLE_FILE; do
-  bal "$BAL_CMD" --offline "$BAL_EXAMPLE_FILE"
+# Loop through examples in the examples directory and execute the command
+echo "Processing examples in the examples directory..."
+cd "$BAL_EXAMPLES_DIR"
+for dir in $(find "$BAL_EXAMPLES_DIR" -type d -maxdepth 1 -mindepth 1); do
```

Shellcheck (0.11.0)

[warning] For loops over find output are fragile. Use find -exec or a while read loop.

(SC2044)